File: Programmers/[77484] 로또의 최고 순위와 최저 순위/SUbbb/77484.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        boolean[] nums = new boolean[46];
        int correctCount = 0;
        int zeroCount = 0;
        
        for (int lotto : lottos) {
            if (lotto == 0) zeroCount++;
            else nums[lotto] = true;
        }
        
        for (int win : win_nums)
            if (nums[win]) correctCount++;
        
        int max = getRanking(correctCount + zeroCount);
        int min = getRanking(correctCount);
        
        return new int[]{max, min};
    }
    
    private int getRanking(int count) {
        if (count <= 1) return 6;
        return 7 - count;
    }
}

File: Programmers/[77484] 로또의 최고 순위와 최저 순위/Johoseong/77484.py
def solution(lottos, win_nums):
    MAX = 0
    MIN = 0

    for l in lottos:
        if l == 0:
            MAX += 1
            continue
        if l in win_nums:
            MAX += 1
            MIN += 1
    if MAX == 0: MAX = 1
    if MIN == 0: MIN = 1
    return [7 - MAX, 7 - MIN]

print(solution([0, 0, 0, 0, 0, 0], [38, 19, 20, 40, 15, 25]))